lib/igniter/project/application.ex

```diff
@@ -26,8 +26,10 @@ defmodule Igniter.Project.Application do
       |> Sourceror.Zipper.zip()
 
     with {:ok, zipper} <- Igniter.Code.Function.move_to_def(zipper, :project, 0),
-         zipper <- Igniter.Code.Common.rightmost(zipper),
-         true <- Igniter.Code.List.list?(zipper),
+         {:ok, zipper} <-
+           Igniter.Code.Common.move_to(zipper, &match?({:__block__, _, [:app]}, &1.node)),
```

Comment:
I think this might be too liberal. Any other option contained that contains an `app` option, anywhere in the structure could be construed as the app config we are looking for, no matter how deep we search. I think we should start simpler, going to the rightmost literal list, and otherwise complaining. We have a `deps_location` pattern for allowing users to configure where in the deps function their list is, so we could eventually do similar patterns here, but this method can yield false *positives* which I think is more dangerous.

Reply:
I was busy when I typed the above, sorry 😆 its not very clear.

The current feature could have a false positive, i.e if someone had something like this:

```elixir
def project() do
  [
    foo: [app: :foo], # <- we'd find this one.
   app: :foo
  ]
end
```

And false positives are dangerous. I think "left most list" is really the only safe heuristic to apply for now.

Reply:
It was clear enough 😄 

Yeah that makes sense, I'm gonna look at that `deps_location` and soon push changes to this branch.